### PR TITLE
Add model and views for bank accounts

### PR DESCRIPTION
**Pull request by paroga**

### CODE REVIEW

**Comment by paroga** [comment]
@wvengen ping?

File: app/models/bank_account.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class BankAccount < ActiveRecord::Base
+
+  has_many :bank_transactions, dependent: :destroy
+
+  normalize_attributes :name, :iban, :description
+
+  validates :name, :presence => true, :uniqueness => true, :length => { :minimum => 2 }
+  validates :iban, :presence => true, :uniqueness => true
+  validates_format_of :iban, :with => /\A[A-Z]{2}[0-9]{2}[0-9A-Z]{,30}\z/
```
Comment (wvengen) [inline app/models/bank_account.rb]: Do we need to validate the check digit?

File: app/views/finance/bank_transactions/show.html.haml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  %b= heading_helper(BankTransaction, :financial_link) + ':'
+  - if @bank_transaction.financial_link
+    = link_to t('ui.show'), finance_link_path(@bank_transaction.financial_link)
+  - else
+    = link_to t('.add_financial_link'), '/'
+
+- unless @bank_transaction.receipt.blank?
+  %pre= preserve @bank_transaction.receipt
+- if @bank_transaction.image
+  %p= image_tag @bank_transaction.image_url
```
Comment (wvengen) [inline app/views/finance/bank_transactions/show.html.haml]: This is a bit ugly for a view. I this is present on other places, but for new views, what about doing a table or a form with static controls? [not a blocking question]

File: config/locales/en.yml [hunk 3]
```diff
@@ -902,6 +903,18 @@ en:
         reload: Reload summary
         with_extra_charge: 'with extra charge:'
         without_extra_charge: 'without extra charge:'
+    bank_accounts:
+      index:
+        account_statement: Account statement
+        title: Bank Accounts
+    bank_transactions:
+      index:
+        search_placeholder: Search ...
```
Comment (wvengen) [inline config/locales/en.yml]: What about moving this to a general i18n section (like cancel etc.)?

File: app/models/bank_account.rb (hunk 1 above)
Comment (paroga) [reply app/models/bank_account.rb]: we don't do it at any of the other places (using a IBAN checkin library is IMHO a independent point)

File: app/views/finance/bank_transactions/show.html.haml (hunk 2 above)
Comment (paroga) [reply app/views/finance/bank_transactions/show.html.haml]: 100% agree, but i don't want to start a new design pattern. i'd like to do this in a more general "make the views beautiful" change, if you don't have something i can copy from

File: config/locales/en.yml (hunk 3 above)
Comment (paroga) [reply config/locales/en.yml]: agree

File: app/views/finance/bank_transactions/show.html.haml (hunk 2 above)
Comment (wvengen) [reply app/views/finance/bank_transactions/show.html.haml]: Ok, that's fair. We have several things here.

Comment (wvengen) [reply app/views/finance/bank_transactions/show.html.haml]: Ok, that's fair. We have several approaches currently with their pros and cons.

File: app/models/bank_account.rb (hunk 1 above)
Comment (wvengen) [reply app/models/bank_account.rb]: Agree.

**Comment by paroga** [comment]
@wvengen i've pushed a new version...

**Pull request merged**